test/e2e/e2e_toomanyrestarts_test.go

```diff
@@ -0,0 +1,190 @@
+			},
+			Spec: v1.PodSpec{
+				Containers: []v1.Container{{
+					Name:            "pause",
+					ImagePullPolicy: "Never",
+					Image:           "kubernetes/pause",
+					Ports:           []v1.ContainerPort{{ContainerPort: 80}},
+				}},
+			},
+		}
+		pod.Status = v1.PodStatus{
+			InitContainerStatuses: []v1.ContainerStatus{
+				{
+					RestartCount: 5 * i,
+				},
+			},
+			ContainerStatuses: []v1.ContainerStatus{
+				{
+					RestartCount: 10 * i,
+				},
+				{
+					RestartCount: 10 * i,
+				},
+			},
+		}
+
+		t.Logf("Creating pod %v in %v namespace for node %v", pod.Name, pod.Namespace, workerNodes[0].Name)
+		_, err := clientSet.CoreV1().Pods(pod.Namespace).Create(ctx, pod, metav1.CreateOptions{})
+		if err != nil {
+			t.Logf("Error creating restart pods: %v", err)
+			if err = clientSet.CoreV1().Pods(pod.Namespace).DeleteCollection(ctx, metav1.DeleteOptions{}, metav1.ListOptions{
+				LabelSelector: labels.SelectorFromSet(labels.Set(map[string]string{"test": "restart-pod", "name": "toomanyrestarts"})).String(),
+			}); err != nil {
+				t.Fatalf("Unable to delete restart pods: %v", err)
+			}
+			return
+		}
+	}
+
+	tests := []struct {
+		name                    string
+		podRestartThreshold     int32
+		includingInitContainers bool
+		expectedEvictedPodCount int
+	}{
+		{
+			name:                    "test-no-evictions",
+			podRestartThreshold:     int32(10000),
+			includingInitContainers: true,
+			expectedEvictedPodCount: 0,
+		},
+		{
+			name:                    "test-three-evictions",
+			podRestartThreshold:     int32(1),
+			includingInitContainers: false,
+			expectedEvictedPodCount: 3,
+		},
+		{
+			name:                    "test-one-evictions",
+			podRestartThreshold:     int32(70),
+			includingInitContainers: true,
+			expectedEvictedPodCount: 1,
+		},
+	}
+	for _, tc := range tests {
+		t.Log("Creating RC with 4 replicas owning the created pods")
+		rc := RcByNameContainer("toomanyrestarts", testNamespace.Name, int32(4), map[string]string{"test": "restart-pod"}, nil, "")
+
+		_, err = clientSet.CoreV1().ReplicationControllers(rc.Namespace).Create(ctx, rc, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			t.Log("RC already exist")
+		} else if err != nil {
+			t.Fatalf("Error creating RC %s %v", tc.name, err)
+		}
+		defer deleteRC(ctx, t, clientSet, rc)
+		waitForRCPodsRunning(ctx, t, clientSet, rc)
+
+		// Run RemovePodsHavingTooManyRestarts strategy
+		evictionPolicyGroupVersion, err := eutils.SupportEviction(clientSet)
+		if err != nil || len(evictionPolicyGroupVersion) == 0 {
+			t.Fatalf("Error creating eviction policy group for %s: %v", tc.name, err)
+		}
+		podEvictor := evictions.NewPodEvictor(
+			clientSet,
+			evictionPolicyGroupVersion,
+			false,
+			0,
+			nodes,
+			true,
+			false,
+			false,
+		)
+
+		podsOnNodeBefore, err := podutil.ListPodsOnANode(ctx, clientSet, workerNodes[0], podutil.WithFilter(podEvictor.Evictable().IsEvictable))
```

Comment:
`workerNodes[0]` is too specific. In different cluster with many worker nodes the pods may be scheduled to different worker nodes than the first one. Can you update the code to list all relevant pods from all the worker nodes instead?

Reply:
Thanks, I check the E2E Test infra, it only create a Kind cluster, no other nodes join, that's why I use workerNode[0], and I also reference other test code, they use the  same word.

Reply:
If you find a way to write the code independent of the number of worker nodes the better. Otherwise, it's not required. Just better to have.